tests/api/buffer_device_address.cpp

```diff
@@ -0,0 +1,105 @@
+// Copyright 2024 The clvk authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "testcl.hpp"
+
+// Extension definitions
+#ifndef cl_ext_buffer_device_address
+#define cl_ext_buffer_device_address 1
+#define CL_DEVICE_PTR_EXT 0xff01
+#define CL_MEM_DEVICE_ADDRESS_EXT (1ul << 31)
+#define CL_MEM_DEVICE_PTR_EXT 0xff01
+typedef cl_ulong cl_mem_device_address_EXT;
+typedef cl_int(CL_API_CALL *clSetKernelArgDevicePointerEXT_fn)(
+    cl_kernel kernel, cl_uint arg_index, cl_mem_device_address_EXT dev_addr);
+#endif
+
+TEST_F(WithCommandQueue, BufferDeviceAddress) {
+    // First check if device supports the extension
+    size_t ext_size;
+    GetDeviceInfo(CL_DEVICE_EXTENSIONS, 0, nullptr, &ext_size);
+    
+    std::vector<char> extensions(ext_size);
+    GetDeviceInfo(CL_DEVICE_EXTENSIONS, ext_size, extensions.data(), nullptr);
+
+    bool hasBufferDeviceAddress = 
+        std::string(extensions.data()).find("cl_ext_buffer_device_address") != std::string::npos;
+
+    if (!hasBufferDeviceAddress) {
+        GTEST_SKIP() << "Device does not support cl_ext_buffer_device_address extension";
+    }
```

Comment:
```suggestion
    // First check if device supports the extension
    REQUIRE_EXTENSION("cl_ext_buffer_device_address");
```

This is being introduced by https://github.com/kpet/clvk/pull/748 for another test.